core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -146,18 +149,47 @@ public void onComplete() {
      */
     @Override
     public boolean tryComplete() {
-        topicPartitionDataFromTryComplete = acquirablePartitions();
+        if (anySharePartitionNoLongerExists()) {
+            return forceComplete();
+        }
+        Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData = acquirablePartitions();
```

Comment:
Hi @junrao , we can do that but then once we are in the `else` of this check -` if (anyTopicIdPartitionHasLogReadError(replicaManagerReadResponse) || isMinBytesSatisfied(topicPartitionData))`, after we have release the partitions lock, we also need to` partitionsToComplete.clear()`. Hence, I've tried to avoid doing this by  assigning `partitionsToComplete` once we are sure that we can do a `forceComplete()`